test/solidity/Facets/WhitelistManagerFacet.t.sol

```diff
@@ -0,0 +1,774 @@
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        whitelistMgr.addToWhitelist(address(c1));
+        address[] memory approved = whitelistMgr.getWhitelistedAddresses();
+        assertEq(approved[0], address(c1));
+
+        vm.stopPrank();
+    }
+
+    function test_SucceedsIfOwnerRemovesAddress() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        vm.expectEmit(true, true, true, true);
+        emit AddressWhitelisted(address(c1));
+
+        whitelistMgr.addToWhitelist(address(c1));
+
+        vm.expectEmit(true, true, true, true);
+        emit AddressRemoved(address(c1));
+
+        whitelistMgr.removeFromWhitelist(address(c1));
+
+        vm.stopPrank();
+
+        address[] memory approved = whitelistMgr.getWhitelistedAddresses();
+        assertEq(approved.length, 0);
+    }
+
+    function test_SucceedsIfOwnerBatchAddsAddresses() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        address[] memory addresses = new address[](3);
+        addresses[0] = address(c1);
+        addresses[1] = address(c2);
+        addresses[2] = address(c3);
+        whitelistMgr.batchAddToWhitelist(addresses);
+        address[] memory approved = whitelistMgr.getWhitelistedAddresses();
+        assertEq(approved[0], addresses[0]);
+        assertEq(approved[1], addresses[1]);
+        assertEq(approved[2], addresses[2]);
+        assertEq(approved.length, 3);
+
+        vm.stopPrank();
+    }
+
+    function test_SucceedsIfOwnerBatchRemovesAddresses() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        address[] memory addresses = new address[](3);
+        addresses[0] = address(c1);
+        addresses[1] = address(c2);
+        addresses[2] = address(c3);
+        whitelistMgr.batchAddToWhitelist(addresses);
+
+        address[] memory remove = new address[](2);
+        remove[0] = address(c1);
+        remove[1] = address(c2);
+        whitelistMgr.batchRemoveFromWhitelist(remove);
+
+        address[] memory approved = whitelistMgr.getWhitelistedAddresses();
+        assertEq(approved.length, 1);
+        assertEq(approved[0], addresses[2]);
+
+        vm.stopPrank();
+    }
+
+    function test_SucceedsIfOwnerApprovesFunctionSelector() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        bytes4 selector = hex"faceface";
+
+        vm.expectEmit(true, true, true, true);
+        emit FunctionSelectorApprovalChanged(selector, true);
+        whitelistMgr.setFunctionApprovalBySelector(selector, true);
+        assertTrue(whitelistMgr.isFunctionApproved(selector));
+
+        vm.stopPrank();
+    }
+
+    function test_SucceedsIfOwnerBatchApprovesFunctionSelectors() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        bytes4[] memory selectors = new bytes4[](5);
+        selectors[0] = bytes4(hex"faceface");
+        selectors[1] = bytes4(hex"deadbeef");
+        selectors[2] = bytes4(hex"deaddead");
+        selectors[3] = bytes4(hex"deadface");
+        selectors[4] = bytes4(hex"beefbeef");
```

Comment:
very creative :)
So that's what you think about when you are coding...beef and dead faces...:)